dftimewolf/lib/collectors/gsheets.py

```diff
@@ -0,0 +1,257 @@
+# -*- coding: utf-8 -*-
+"""Pulls entries from Google Sheets."""
+
+
+import os.path
+import re
+import tempfile
+from typing import List
+from typing import Optional
+
+import filelock
+import pandas as pd
+from google.auth.exceptions import DefaultCredentialsError
+from google.auth.exceptions import RefreshError
+from google.auth.transport.requests import Request
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient import discovery
+from pandas.core.frame import DataFrame
+
+from dftimewolf.lib import module
+from dftimewolf.lib.containers import containers
+from dftimewolf.lib.modules import manager as modules_manager
+from dftimewolf.lib.state import DFTimewolfState
+
+
+class GoogleSheetsCollector(module.BaseModule):
+  """Collector for entries from Google Sheets."""
+
+  SCOPES = ['https://www.googleapis.com/auth/spreadsheets.readonly']
+
+  _CREDENTIALS_FILENAME = '.dftimewolf_google_sheets_credentials.json'
+  _CLIENT_SECRET_FILENAME = '.dftimewolf_google_sheets_client_secret.json'
+
+  def __init__(self,
+               state: DFTimewolfState,
+               name: Optional[str] = None,
+               critical: bool = False) -> None:
+    """Initializes a Google Sheets collector."""
+    super(GoogleSheetsCollector, self).__init__(
+        state, name=name, critical=critical)
+    self._credentials = None
+    self._spreadsheet_id = ''
+    self._sheets_names: List[str] = []
+    # These are mandatory columns required by Timesketch.
+    self._mandatory_columns = ['message', 'datetime', 'timestamp_desc']
+    self._all_sheets = False
+    self._validate_columns = True
+
+  # pylint: disable=arguments-differ
+  def SetUp(self,
+            spreadsheet: str,
+            sheets_names: List[str],
+            validate_columns: bool = True) -> None:
+    """Sets up a a Google Sheets collector.
+
+    Args:
+      spreadsheet: ID or URL of the sheet to pull data from
+      sheets_names: List of sheets names inside the spreadsheet to parse. 'All'
```

Comment:
sheet_names: Fixed.

All: Ya valid point, doesn't make a difference, I rather keep things explicit better for readability i think.

Reply:
Imo if the doc string and the argument says that default is all worksheets that doesn't really hurt the readability, and avoids having to work with an 'all' string which I think we should avoid if we can. If you do decide to keep it in please also update this doc string to 'all' lowercase to avoid confusion.

Reply:
I will remove "All" just for you @itsmvd  ;)